sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,504 @@
+
+const WICKED_CONFIG_DIR: &str = "/etc/wicked/ifconfig";
+const WICKED_FILE_EXT: &str = "xml";
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename = "interface")]
+pub(crate) struct WickedInterface {
+    name: InterfaceName,
+    control: WickedControl,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "ipv4:dhcp")]
+    ipv4_dhcp: Option<WickedDhcp4>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "ipv6:dhcp")]
+    ipv6_dhcp: Option<WickedDhcp6>,
+}
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+struct WickedControl {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    mode: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    link_detection: Option<LinkDetection>,
+    // TODO: `serde_xml_rs` has a known issue with serializing nested structures, where it will
+    // insert additional tag with the structure name.  It has since been fixed but not released
+    // officially yet.  This struct member works around that issue.
+    // https://github.com/RReverser/serde-xml-rs/issues/126
+    // The workaround:
+    // https://stackoverflow.com/questions/70124048/how-to-create-xml-from-struct-in-rust
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+struct LinkDetection {
+    // This will serialize to an empty tag, which works in place of the self-closing tag we have
+    // today
+    require_link: (),
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct WickedDhcp4 {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    route_priority: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    defer_timeout: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flags: Option<AddrConfFlags>,
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct WickedDhcp6 {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    defer_timeout: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    flags: Option<AddrConfFlags>,
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+// This is technically an enum, but considering we don't expose anything other than "optional" to
+// the user, a struct makes handling tags much simpler.
+#[derive(Default, Clone, Debug, Serialize, PartialEq)]
+struct AddrConfFlags {
+    optional: (),
+    #[serde(flatten, skip)]
+    _f: (),
+}
+
+// We assume that all configured interfaces are wanted at boot and will require a link to
+// be considered configured
+impl Default for WickedControl {
```

Comment:
This file might be easier to follow if the `impls` are near the structs they belong to.